devicetypes/qubino/qubino-flush-dimmer-0-10V.src/qubino-flush-dimmer-0-10V.groovy

```diff
@@ -0,0 +1,671 @@
+            [
+                completedSetup: true,
+                label: childComponentLabel,
+                isComponent: false
+            ])
+        log.debug "createChildDevice: ${childDevice}"
+        childDevice
+    } catch(Exception e) {
+        log.debug "Exception: ${e}"
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStartLevelChange cmd) {
+	log.debug "SwitchMultilevelStartLevelChange: ${cmd}"
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStopLevelChange cmd) {
+	log.debug "SwitchMultilevelStopLevelChange: ${cmd}"
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.debug "manufacturerId: $cmd.manufacturerId"
+	log.debug "manufacturerName: $cmd.manufacturerName"
+	log.debug "productId: $cmd.productId"
+	log.debug "productTypeId: $cmd.productTypeId"
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	updateDataValue("MSR", msr)
+	updateDataValue("manufacturer", cmd.manufacturerName)
+	createEvent([descriptionText: "$device.displayName MSR: $msr", isStateChange: false])
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	// Handles other Z-Wave commands that are not supported here
+	log.debug "Command: ${cmd}"
+	[:]
+}
+
+def on() {
+	delayBetween([
+			zwave.basicV1.basicSet(value: 0xFF).format(),
+			zwave.switchMultilevelV3.switchMultilevelGet().format()
+	], 5000)
+}
+
+def off() {
+	delayBetween([
+			zwave.basicV1.basicSet(value: 0x00).format(),
+			zwave.switchMultilevelV3.switchMultilevelGet().format()
+	], 5000)
+}
+
+def setLevel(value) {
+	log.debug "setLevel >> value: $value"
+	def valueaux = value as Integer
+	def level = Math.max(Math.min(valueaux, 99), 0)
+	delayBetween([zwave.basicV1.basicSet(value: level).format(), zwave.switchMultilevelV3.switchMultilevelGet().format()], 5000)
+}
+
+def setLevel(value, duration) {
+	log.debug "setLevel >> value: $value, duration: $duration"
+	def valueaux = value as Integer
+	def level = Math.max(Math.min(valueaux, 99), 0)
+	def dimmingDuration = duration < 128 ? duration : 128 + Math.round(duration / 60)
+	def getStatusDelay = duration < 128 ? (duration * 1000) + 2000 : (Math.round(duration / 60) * 60 * 1000) + 2000
+	delayBetween([zwave.switchMultilevelV3.switchMultilevelSet(value: level, dimmingDuration: dimmingDuration).format(),
+				  zwave.switchMultilevelV3.switchMultilevelGet().format()], getStatusDelay)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh"
+	def commands = []
+
+	commands << zwave.basicV1.basicGet().format()
+	commands << zwave.switchMultilevelV3.switchMultilevelGet().format()
+	commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 0).format()
```

Comment:
What is this get for? Aren't all the sensors on endpoints? Wouldn't this need to be encapsulated or only sent conditionally?